### DISCUSSION TITLE
Optimize images inside SVG

### DISCUSSION THREAD

**Discussion by florianbepunkt** [vercel/next.js#29948] [Ideas]
### Describe the feature you'd like to request

We use svg image masks a lot. We would like to make use of the next.js image otpimization for `<images>` inside an `<svg>` tag.

```ts
<svg>
<image /> // optimize this
</svg>
```

full example:

```ts
import BackgroundImage from "../../public/typo-effects/mvn-typo-bg.jpg";

<svg viewBox="0 0 475.5 450" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
      <defs>
        <clipPath id="feature-one-clip-path-mobile">
          <path d="somePath" fill="none" />
        </clipPath>
      </defs>
        <g clipPath="url(#feature-one-clip-path-mobile)">
          <image // this should be optimized
            height="450.269"
            transform="translate(-40.185 -2.538)"
            width="516.723"
            xlinkHref={BackgroundImage.src}
          />
        </g>
    </svg>
```

### Describe the solution you'd like

Ability to use the next.js image component or the same optimization techniques here.

### Describe alternatives you've considered

I have none.

**Comment by andersonszdc** [comment]
Yeaah, I really need this!!!

**Comment by spennydix** [comment]
Same! Any website that clips or masks images inside of svg would benefit!

**Comment by anarkrypto** [comment]
I solved it using the **clipPath** in the parent div and adding **objectFit: cover** to the Next.js Image

Here's a demo with **resize support**:

https://codesandbox.io/p/sandbox/shapednextjsimage-s4pl4t?file=%2Fapp%2Fpage.tsx%3A18%2C25

![image](https://github.com/vercel/next.js/assets/32111208/3f602c2f-305e-411c-8c6e-4d1a7bfa8595)
